checker/src/main/java/org/checkerframework/checker/signedness/jdk.astub

```diff
@@ -3268,6 +3268,34 @@ class PrintStream {
 class RandomAccessFile {
     public void write(@PolySigned byte b[]);
     public void write(@PolySigned byte b[], int off, int len);
+    public void write(@PolySigned int b) throws IOException;
+    public final void writeByte(@PolySigned int v) throws IOException;
+    public final void writeShort(@PolySigned int v) throws IOException;
+    public final void writeChar(@PolySigned int v) throws IOException;
+    public final void writeInt(@PolySigned int v) throws IOException;
+    public final void writeLong(@PolySigned long v) throws IOException;
+    private native void writeBytes(@PolySigned byte b[], int off, int len) throws IOException;
+    public final void writeChars(String s) throws IOException {
+        int clen = s.length();
+        int blen = 2*clen;
+        @Unsigned byte[] b = new byte[blen];
+        char[] c = new char[clen];
+        s.getChars(0, clen, c, 0);
+        for (int i = 0, j = 0; i < clen; i++) {
+            b[j++] = (byte)(c[i] >>> 8);
+            b[j++] = (byte)(c[i] >>> 0);
+        }
+        writeBytes(b, 0, blen);
+    }
+    public native @PolySigned int read() throws IOException;
+    private native int readBytes(@PolySigned byte b[], int off, int len) throws IOException;
+    public int read(@PolySigned byte[] b) throws IOException;
+    public int read(@PolySigned byte b[], int off, int len) throws IOException;
+    public void seek(@Unsigned long pos) throws IOException;
```

Comment:
This annotation looks wrong to me.  The argument is interpreted as signed.  In particular, passing
9223372036854775808 is illegal because it is interpreted as a negative number.

The value does need to be positive, but that is orthogonal to how the argument is interpreted, which is what `@Unsigned` is about.

Reply:
Yes, I got your point. It looks wrong to me too, but it in the case study done by me the parameter which was getting passed was ```@Unsigned```, so will we always have to suppress this kind of error whenever it occurs.

Reply:
In the case study code, the value should be in the positive integer range.  Not every unsigned value is permitted.  So it sounds like the case study code might be wrong (it could pass wrong values), or there are properties you need to add to it, such as an [`@IntRange`](https://checkerframework.org/api/org/checkerframework/common/value/qual/IntRange.html) annotation.

You should never write an incorrect annotation in a library just to suppress a warning in some particular client.  Incorrect library annotations make the checker unsound for every client program, which is a bad tradeoff.

Reply:
Thanks for the prompt reply. I got it.